x/incentives/types/msgs.go

```diff
@@ -138,3 +139,53 @@ func (m MsgAddToGauge) GetSigners() []sdk.AccAddress {
 	owner, _ := sdk.AccAddressFromBech32(m.Owner)
 	return []sdk.AccAddress{owner}
 }
+
+var _ sdk.Msg = &MsgCreateGroup{}
+
+// NewMsgCreateGroup creates a message to create a group with the provided parameters.
+func NewMsgCreateGroup(rewards sdk.Coins, numEpochsPaidOver uint64, owner sdk.AccAddress, poolIds []uint64) *MsgCreateGroup {
+	return &MsgCreateGroup{
+		Coins:             rewards,
+		NumEpochsPaidOver: numEpochsPaidOver,
+		Owner:             owner.String(),
+		PoolIds:           poolIds,
+	}
+}
+
+// Route takes a create group message, then returns the RouterKey.
+func (m MsgCreateGroup) Route() string { return RouterKey }
+
+// Type takes a create group message, then returns the message type.
+func (m MsgCreateGroup) Type() string { return TypeMsgCreateGroup }
+
+// ValidateBasic checks that the create group message is valid.
+func (m MsgCreateGroup) ValidateBasic() error {
+	if m.Owner == "" {
+		return errors.New("owner should be set")
+	}
+	if len(m.PoolIds) < 2 {
+		return errors.New("pool ids should be composed of at least 2 pool IDs")
+	}
+
+	// Check for uniqueness
+	poolIdsMap := make(map[uint64]bool)
+	for _, id := range m.PoolIds {
+		if _, exists := poolIdsMap[id]; exists {
+			return errors.New("pool ids should be unique")
+		}
+		poolIdsMap[id] = true
+	}
```

Comment:
Could we extract this into `osmoutils`? We could then reuse it in `CreateGroup` in that case

Reply:
extracted to osmoassert here https://github.com/osmosis-labs/osmosis/pull/6588/commits/c622e9b7b17132e7ec5161202bac682bb857b714